### PR TITLE
{AKS} Support to enable azure container insight monitoring when the sku name is automatic

### PR DESCRIPTION
**Pull request by xmzhao0822**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az aks create -g xmrg -n xmmc --sku automatic

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -3166,6 +3166,7 @@ def test_aks_automatic_sku(self, resource_group, resource_group_location):
             "AKSHTTPCustomFeatures=Microsoft.ContainerService/EnableAPIServerVnetIntegrationPreview,"
             "AKSHTTPCustomFeatures=Microsoft.ContainerService/SafeguardsPreview,"
             "AKSHTTPCustomFeatures=Microsoft.ContainerService/NRGLockdownPreview,"
+            "AKSHTTPCustomFeatures=Microsoft.ContainerService/AKS-PrometheusAddonPreview,"
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3169]: Please mark the case as `@live_only()` as the workspace id is not mocked properly so the test could not be replayed.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -317,6 +317,20 @@ def get_sku_name(self) -> str:
                 skuName = CONST_MANAGED_CLUSTER_SKU_NAME_BASE
         return skuName
 
+    def _get_enable_addons(self, enable_validation: bool = False) -> List[str]:
+        enable_addons = super()._get_enable_addons()
+        sku_name = self.raw_param.get("sku")
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:322]: I think `self.get_sku_name()` would be better than `self.raw_param.get("sku")`

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -317,6 +317,20 @@ def get_sku_name(self) -> str:
...
+        if sku_name == CONST_MANAGED_CLUSTER_SKU_NAME_AUTOMATIC:
+            enable_addons.append("monitoring")
+        return enable_addons
+
+    def get_enable_msi_auth_for_monitoring(self) -> Union[bool, None]:
+        enable_msi_auth_for_monitoring = super().get_enable_msi_auth_for_monitoring()
+        sku_name = self.raw_param.get("sku")
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:329]: same here

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=92256475&view=results) to validate the change.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (xmzhao0822) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3169]: Marked the case as @live_only() with comments

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 2 above)
Comment (xmzhao0822) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: Updated to use self.get_sku_name()

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 3 above)
Comment (xmzhao0822) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: Updated to use self.get_sku_name()

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**